test/e2e/Wrappers.go

```diff
@@ -0,0 +1,143 @@
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package e2e
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/apache/yunikorn-core/pkg/common/configs"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/configmanager"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/common"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/k8s"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/yunikorn"
+)
+
+var k = k8s.KubeCtl{}
+
+func UpdateConfigMapWrapper(oldConfigMap *v1.ConfigMap, schedPolicy string, annotation string) {
+	By("Enabling new scheduling config")
+
+	// Save old configMap
+	Ω(k.SetClient()).To(BeNil())
+	var c, err = k.GetConfigMaps(configmanager.YuniKornTestConfig.YkNamespace,
+		configmanager.GetConfigMapName())
+	Ω(err).NotTo(HaveOccurred())
+	Ω(c).NotTo(BeNil())
+	c.DeepCopyInto(oldConfigMap)
+	Ω(c).Should(BeEquivalentTo(oldConfigMap))
+
+	// Create fair configMap with timestamp
+	sc := common.CreateBasicConfigMap()
+	if schedPolicy != "" {
+		err = common.SetSchedulingPolicy(sc, "default", "root", schedPolicy)
+		Ω(err).NotTo(HaveOccurred())
+	}
+	ts, tsErr := common.SetQueueTimestamp(sc, "default", "root")
+	Ω(tsErr).NotTo(HaveOccurred())
+	configStr, yamlErr := common.ToYAML(sc)
+	Ω(yamlErr).NotTo(HaveOccurred())
+	c.Data[configmanager.DefaultPolicyGroup] = configStr
+	var d, err3 = k.UpdateConfigMap(c, configmanager.YuniKornTestConfig.YkNamespace)
+	Ω(err3).NotTo(HaveOccurred())
+	Ω(d).NotTo(BeNil())
+
+	// Updating scheduler pod annotation to trigger force refresh of configmaps
+	// https://jira.cloudera.com/browse/COMPX-4042
+	Ω(k.UpdateYunikornSchedulerPodAnnotation(annotation)).NotTo(HaveOccurred())
+	err = yunikorn.WaitForQueueTS("root", ts, 2*time.Minute)
+	Ω(err).NotTo(HaveOccurred())
+}
+
+func RestoreConfigMapWrapper(oldConfigMap *v1.ConfigMap, annotation string) {
+	Ω(k.SetClient()).To(BeNil())
+	By("Restoring the old config maps")
+	var c, err = k.GetConfigMaps(configmanager.YuniKornTestConfig.YkNamespace,
+		configmanager.GetConfigMapName())
+	Ω(err).NotTo(HaveOccurred())
+	Ω(c).NotTo(BeNil())
+
+	oldSC := new(configs.SchedulerConfig)
+	err = yaml.Unmarshal([]byte(oldConfigMap.Data[configmanager.DefaultPolicyGroup]), oldSC)
+	Ω(err).NotTo(HaveOccurred())
+	ts, tsErr := common.SetQueueTimestamp(oldSC, "default", "root")
+	Ω(tsErr).NotTo(HaveOccurred())
+	c.Data[configmanager.DefaultPolicyGroup], err = common.ToYAML(oldSC)
+	Ω(err).NotTo(HaveOccurred())
+
+	var e, err3 = k.UpdateConfigMap(c, configmanager.YuniKornTestConfig.YkNamespace)
+	Ω(err3).NotTo(HaveOccurred())
+	Ω(e).NotTo(BeNil())
+	// Updating scheduler pod annotation to trigger force refresh of configmaps
+	// https://jira.cloudera.com/browse/COMPX-4042
+	Ω(k.RemoveYunikornSchedulerPodAnnotation(annotation)).NotTo(HaveOccurred())
```

Comment:
This should not be needed anymore with the new config map listener
Internal jira reference must be removed